src/Microsoft.Azure.WebJobs.Host/Scale/DefaultConcurrencyThrottleManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    internal class DefaultConcurrencyThrottleManager : IConcurrencyThrottleManager
+    {
+        private readonly TimeSpan _minUpdateInterval = TimeSpan.FromMilliseconds(1000);
+        private readonly IEnumerable<IConcurrencyThrottleProvider> _throttleProviders;
+        private readonly ILogger _logger;
+
+        private ThrottleState _throttleState;
+        private List<string> _enabledThrottles;
+        private int _consecutiveCount;
+
+        public DefaultConcurrencyThrottleManager(IEnumerable<IConcurrencyThrottleProvider> throttleProviders, ILoggerFactory loggerFactory)
+        {
+            _throttleProviders = throttleProviders;
```

Comment:
This internal type is only created by DI (other than my tests), so the input collection won't change once resolved by DI. However, I can add the ToList because it won't hurt.